adagrams/game.py

```diff
@@ -1,13 +1,162 @@
+    letter_pool.extend(["N"] * 6)
+    letter_pool.extend(["O"] * 8)
+    letter_pool.extend(["P"] * 2)
+    letter_pool.extend(["Q"] * 1)
+    letter_pool.extend(["R"] * 6)
+    letter_pool.extend(["S"] * 4)
+    letter_pool.extend(["T"] * 6)
+    letter_pool.extend(["U"] * 4)
+    letter_pool.extend(["V"] * 2)
+    letter_pool.extend(["W"] * 2)
+    letter_pool.extend(["X"] * 1)
+    letter_pool.extend(["Y"] * 2)
+    letter_pool.extend(["Z"] * 1)
+    return letter_pool
+
 
 def draw_letters():
-    pass
+    #Create the data structure containing the letters and letter distribution.
+    letter_pool = create_letter_pool()
+
+    #Randomly select 10 characters from the data structure
+    letter_hand = []
+    for i in range(10):
+        #Generates a random integer between 0 and the last element in letter_pool list.
+        random_index = randint(0,len(letter_pool)-1)
+        #Access an element within letter_pool list, determined by random_index.
+        random_letter = letter_pool[random_index]
+        #Add the newly generated random letter to letter_hand list.
+        letter_hand.append(random_letter)
+        #Remove the random_letter from letter_pool
+        letter_pool.remove(random_letter)
+
+    #Return letter_hand list of 10 letters.
+    return letter_hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    #Create a dictionary for word to store the word and its frequency.
+    word_count = {}
+    #Iterate through each element in word and add it to word_count dictionary.
+    for char in word:
+        if char in word_count:
+            word_count[char] += 1
+        else:
+            word_count[char] = 1
+    
+
+    #Create a dictionary for letter_bank to store each word and its frequency.
+    letter_bank_count = {}
+    #Iterate through each element in letter_bank list and add it to letter_bank_count dictionary.
+    for i in letter_bank:
+        if i in letter_bank_count:
+            letter_bank_count[i] += 1
+        else:
+            letter_bank_count[i] = 1
+
+    
+    #For every letter in word_count, the letter in letter_bank_count is greater than or equal to it.
+    #Iterate through word_count dictionary
+    for char in word_count.keys():
+        if char not in letter_bank_count or word_count[char] > letter_bank_count[char]:
+            return False
+    return True
+        
+
 
 def score_word(word):
-    pass
+    word = word.upper()
+    #Create a dictionary to store score chart.
+    letter_values = {
+        'A': 1,
+        'B': 3,
+        'C': 3,
+        'D': 2,
+        'E': 1,
+        'F': 4,
+        'G': 2,
+        'H': 4,
+        'I': 1,
+        'J': 8,
+        'K': 5,
+        'L': 1,
+        'M': 3,
+        'N': 1,
+        'O': 1,
+        'P': 3,
+        'Q': 10,
+        'R': 1,
+        'S': 1,
+        'T': 1,
+        'U': 1,
+        'V': 4,
+        'W': 4,
+        'X': 8,
+        'Y': 4,
+        'Z': 10
+    }
```

Comment:
Since this variable isn't changing or being modified, we want to use the python convention of naming it in all uppercase. This signals to other developers that this piece of data should not be modified or re-assigned. 
```suggestion
    LETTER_VALUES = {
    "A": 1, "E": 1, "I": 1, "O": 1, "U": 1,
    "L": 1, "N": 1, "R": 1, "S": 1, "T": 1,
    "D": 2, "G": 2,
    "B": 3, "C": 3, "M": 3, "P": 3,
    "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4,
    "K": 5,
    "J": 8, "X": 8,
    "Q": 10, "Z": 10
	}
```